genty/genty.py

```diff
@@ -306,6 +306,15 @@ def _build_final_method_name(
     if not dataset_name and not repeat_suffix:
         return '{0}{1}'.format(method_name, suffix)
 
+    if dataset_name:
+        # Nosetest multi-processing code parses the full test name
+        # to discern package/module names. Thus any periods in the test-name
+        # causes that code to fail. So replace any periods with the unicode
+        # middle-dot character. Yes, this change is applied independent
+        # of the test runner being used... and that's fine since there is
+        # no real contract as to how the fabricated tests are named.
+        dataset_name = dataset_name.replace('.', '\xb7')
```

Comment:
Consider extracting \xb7 to a class variable, so it's available for tests and possibly for changing in case someone doesn't want this behavior.